integrationtest/backend/tck/src/main/java/org/hibernate/search/integrationtest/backend/tck/testsupport/util/TestedFieldStructure.java

```diff
@@ -89,4 +89,9 @@ public boolean isSingleValued() {
 	public boolean isMultiValued() {
 		return !isSingleValued();
 	}
+
+	public boolean isMultiValuedInRoot() {
+		// all nested object mappings made by tests are multivalued
+		return isMultiValued() || isInNested();
```

Comment:
That's true in `IndexFieldDescriptorIT`, but not everywhere. See the comment in that test:

```
				// In this specific test, we decided that nested fields are multi-valued, while flattened fields are not.
				// That's just a detail though: the two characteristics are not usually related.
```

Maybe you shouldn't change `IndexFieldLocation`/`TestFieldStructure`, and move the condition to `IndexFieldDescriptorIT`?

Reply:
Right, it does make sense to not change `TestedFieldStructure` but only `TestedFieldStructure`.